### PR TITLE
Migration of test_rebalance_new_node.py from glusto to redant

### PR DESCRIPTION
**Pull request by nishith-vihar**
Migration of [test_rebalance_new_node.py](https://github.com/gluster/glusto-tests/blob/master/tests/functional/glusterd/test_rebalance_new_node.py) from glusto to redant
    
Fixes: #292
Fixes: #492
Signed-off-by: Nishith Vihar Sakinala <nsakinal@redhat.com>

### CODE REVIEW

**Comment by nishith-vihar** [comment]
```
test_rebalance_new_node-Generic PASS

Summary:
+---------------+-------+--------------+
|    Category   | Cases | Pass Percent |
+---------------+-------+--------------+
| nonDisruptive |   0   |      0       |
|   Disruptive  |   1   |    100.0     |
|     Total     |   1   |    100.0     |
+---------------+-------+--------------+
Table:
 test_rebalance_new_node
 
+-------------+-------------+-------------------+
| Volume Type | Test Result |  Time taken (sec) |
+-------------+-------------+-------------------+
|   Generic   |     PASS    | 77.74245142936707 |
+-------------+-------------+-------------------+



Framework runtime : 82.07236862182617
```

File: tests/functional/glusterd/test_rebalance_new_node.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        redant.add_brick(self.volume_name1, self.server_list[0],
+                         self.vol_type_inf[self.conv_dict['dist']],
+                         self.server_list, self.brick_roots, True)
+
+        redant.rebalance_start(self.volume_name1, self.server_list[0])
+
+        # peer probe a new node from existing cluster
+        redant.peer_probe(self.server_list[2], self.server_list[0])
+
+        redant.get_rebalance_status(self.volume_name1, self.server_list[0])
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_rebalance_new_node.py:90]: Check for `None` here.

**Comment by srijan-sivakumar** [comment]
@nishith-vihar I'm changing how add brick works...will be pushing a new patch in a couple of minutes...because then you might need to change how add brick is done here..after rebasing the code..

**Comment by srijan-sivakumar** [comment]
or...we can push this change...and I'll modify this test case in my patch..what do you say ?

File: tests/functional/glusterd/test_rebalance_new_node.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+from tests.d_parent_test import DParentTest
+
+# disruptive;
+
+
+class TestCase(DParentTest):
+
+    def terminate(self):
+        try:
+            self.redant.cleanup_volume(self.volume_name1, self.server_list[0])
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_rebalance_new_node.py:33]: Add a `wait_for_io_to_complete` function here, check other TCs for reference.

**Review by nik-redhat — CHANGES_REQUESTED** [review]

**Comment by nik-redhat** [comment]
> or...we can push this change...and I'll modify this test case in my patch..what do you say ?

Let's prefer the former, or else your patch will grow exponentially :joy:

**Comment by srijan-sivakumar** [comment]
> > or...we can push this change...and I'll modify this test case in my patch..what do you say ?
> 
> Let's prefer the former, or else your patch will grow exponentially joy

I solemnly swear that it's not going to exceed 150-200 lines :p

**Comment by nik-redhat** [comment]
> > > or...we can push this change...and I'll modify this test case in my patch..what do you say ?
> > 
> > 
> > Let's prefer the former, or else your patch will grow exponentially joy
> 
> I solemnly swear that it's not going to exceed 150-200 lines :p

Then, it's up to you :p

**Comment by srijan-sivakumar** [comment]
> > > > or...we can push this change...and I'll modify this test case in my patch..what do you say ?
> > > 
> > > 
> > > Let's prefer the former, or else your patch will grow exponentially joy
> > 
> > 
> > I solemnly swear that it's not going to exceed 150-200 lines :p
> 
> Then, it's up to you :p

hehe. Seriously...it won't exceed the limit :D

**Comment by nishith-vihar** [comment]
```
test_rebalance_new_node-Generic PASS

Summary:
+---------------+-------+--------------+
|    Category   | Cases | Pass Percent |
+---------------+-------+--------------+
| nonDisruptive |   0   |      0       |
|   Disruptive  |   1   |    100.0     |
|     Total     |   1   |    100.0     |
+---------------+-------+--------------+
Table:
 test_rebalance_new_node
 
+-------------+-------------+-------------------+
| Volume Type | Test Result |  Time taken (sec) |
+-------------+-------------+-------------------+
|   Generic   |     PASS    | 74.77417588233948 |
+-------------+-------------+-------------------+



Framework runtime : 79.0829963684082
```

**Review by srijan-sivakumar — APPROVED** [review]
LGTM. Taking it in.

**Pull request merged**